patches/minecraft/net/minecraft/advancements/critereon/ItemPredicate.java.patch

```diff
@@ -0,0 +1,10 @@
+--- ../src-base/minecraft/net/minecraft/advancements/critereon/ItemPredicate.java
++++ ../src-work/minecraft/net/minecraft/advancements/critereon/ItemPredicate.java
+@@ -104,6 +104,7 @@
+         if (p_192492_0_ != null && !p_192492_0_.isJsonNull())
+         {
+             JsonObject jsonobject = JsonUtils.func_151210_l(p_192492_0_, "item");
++            if (jsonobject.has("type")) { final ResourceLocation rl = new ResourceLocation(JsonUtils.func_151200_h(jsonobject, "type")); final Map<ResourceLocation, java.util.function.Function<JsonObject, ItemPredicate>> map = net.minecraftforge.advancements.critereon.ItemPredicates.getPredicates(); if (map.containsKey(rl)) return map.get(rl).apply(jsonobject); else throw new JsonSyntaxException("There is no ItemPredicate of type "+rl); }
```

Comment:
we want small patches, but they should still be readable. Some newlines would be acceptable here I think.